sklearn/decomposition/_pca.py

```diff
@@ -851,4 +851,5 @@ def __sklearn_tags__(self):
         tags = super().__sklearn_tags__()
         tags.transformer_tags.preserves_dtype = ["float64", "float32"]
         tags.array_api_support = True
+        tags.input_tags.sparse = True
```

Comment:
Here, it will depends on the solver:


```suggestion
        tags.input_tags.sparse = self.svd_solver in ("auto", "arpack", "covariance_eigh")
```

I just realise that `score_samples` does not accept at all sparse data. We might check if it is still the case with the improvement recently done with the `"covariance_eigh"` solver or if we can go around the original issue.